sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/SettingSelector.java

```diff
@@ -68,18 +70,19 @@ public SettingSelector() {
      * <p>
      * Examples:
      * <ul>
-     * <li>If {@code keys = "*"}, settings with any key are returned.</li>
-     * <li>If {@code keys = "abc1234"}, settings with a key equal to "abc1234" are returned.</li>
-     * <li>If {@code keys = "abc*"}, settings with a key starting with "abc" are returned.</li>
-     * <li>If {@code keys = "*abc*"}, settings with a key containing "abc" are returned.</li>
-     * <li>If {@code keys = "abc,def"}, settings with a key equal to "abc" or "def" are returned.</li>
+     * <li>If {@code keyFilter = "*"}, settings with any key are returned.</li>
+     * <li>If {@code keyFilter = "abc1234"}, settings with a key equal to "abc1234" are returned.</li>
+     * <li>If {@code keyFilter = "abc*"}, settings with a key starting with "abc" are returned.</li>
+     * <li>If {@code keyFilter = "*abc*"}, settings with a key containing "abc" are returned.</li>
+     * <li>If {@code keyFilter = "abc,def"}, settings with a key equal to "abc" or "def" are returned.</li>
      * </ul>
      *
-     * @param keys The expressions to filter ConfigurationSetting keys on.
+     * @param keyFilter The expressions to filter ConfigurationSetting keys on.
      * @return The updated SettingSelector object
      */
-    public SettingSelector setKeys(String... keys) {
-        this.keys = keys;
+    public SettingSelector setKeyFilter(String keyFilter) {
+        Objects.requireNonNull(keyFilter);
```

Comment:
Add a message when using Objects.requireNonNull or it'll throw a NullPointer with no indication of what happened.
Also, update the javadocs with `@throws`.

Reply:
Actually, looking at our SDK guidelines, we shouldn't be doing this verification at all and allowing the service to return a bad request.
https://azure.github.io/azure-sdk/general_implementation.html#parameter-validation

Reply:
Ahh. That is correct. Thanks